### DISCUSSION TITLE
Retrieve instances of recurring events.

### DISCUSSION THREAD

**Discussion by ghost** [spatie/icalendar-generator#60] [Q&A]
Hey folks,

I'm using the Spatie icalendar-generator to implement an iCal feed in a client's CRM tool. I've adopted RRules to make recurring events rather than duplicating events. Now I wonder if, somehow, I can get a list of all dates and times an event recurs. 

Thank you in advance for your reply and thinking along!

**Comment by rubenvanassche** [comment]
Hi @DemienD,

I think you'll probably gonna need a package like this one: https://github.com/rlanvin/php-rrule. We do not support it at the moment since the purpose of this package is only to make the feeds not trying to read them.
